src/xunit.analyzers/Analysis/XunitContext.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+
+namespace Xunit.Analyzers
+{
+    public class XunitContext
+    {
+        readonly Lazy<AbstractionsContext> abstractions;
+        readonly Lazy<CoreContext> core;
+        readonly Lazy<ExecutionContext> execution;
+        readonly Lazy<bool> hasAbstractionsReference;
+        readonly Lazy<bool> hasCoreReference;
+        readonly Lazy<bool> hasExecutionReference;
+
+        internal XunitContext(Compilation compilation, XunitCapabilitiesFactory capabilitiesFactory)
+        {
+            Capabilities = capabilitiesFactory(compilation);
+            Compilation = compilation;
+
+            hasAbstractionsReference = new Lazy<bool>(() => Compilation.ReferencedAssemblyNames.FirstOrDefault(x => x.Name == "xunit.abstractions") != null);
```

Comment:
nit: Why not just `Any(x => x.Name == "xunit.abstractions")`?

Reply:
Because I'm an idiot and didn't think of it. 😝